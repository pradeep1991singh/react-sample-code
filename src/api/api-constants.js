export const API_BASE_URL = 'http://localhost:8080/api';
export const MENU_ITEMS_API = '/menu-items';
export const SEARCH_API = '/menu-items/search';
