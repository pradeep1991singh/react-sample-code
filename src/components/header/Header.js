import React, { Component } from 'react';

import NavigationBar from '../NavigationBar';
import './header.css';

class Header extends Component {
  render() {
    return <NavigationBar onMenuClick={this.props.onMenuClick} />;
  }
}

export default Header;
