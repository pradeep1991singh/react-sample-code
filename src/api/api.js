import { API_BASE_URL, MENU_ITEMS_API, SEARCH_API } from './api-constants';

// fetch drop down items
export const getDropDownItems = () => {
  return fetch(API_BASE_URL + MENU_ITEMS_API);
};

// search api call
export const getSearchResults = searchText =>
  fetch(`${API_BASE_URL}${SEARCH_API}?searchText=${searchText}`);
