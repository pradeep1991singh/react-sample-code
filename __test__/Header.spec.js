import React from 'react';
import renderer from 'react-test-renderer';
import { getDropDownItems, getSearchResults } from '../src/api/api';
import Header from '../src/components/NavigationBar';

const props = {
  onMenuClick: jest.fn()
};

describe('Drop down menus', () => {
  it('should render drop down menu', () => {
    const component = renderer.create(<Header {...props} />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
