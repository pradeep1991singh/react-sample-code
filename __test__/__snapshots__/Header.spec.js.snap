// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drop down menus should render drop down menu 1`] = `
<nav
  className="navbar navbar-default"
>
  <div
    className="container"
  >
    <div
      className="navbar-header"
    >
      <a
        className="navbar-brand"
        href="/"
      >
        <img
          alt="logo"
          className="App-logo"
          src="test-file-stub"
        />
      </a>
    </div>
    <ul
      className="nav navbar-nav"
      role={null}
    >
      <li
        className="dropdown"
        style={undefined}
      >
        <a
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle"
          href="#"
          id="basic-nav-dropdown"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          Tags
           
          <span
            className="caret"
          />
        </a>
        <ul
          aria-labelledby="basic-nav-dropdown"
          className="dropdown-menu"
          role="menu"
        >
          <li
            className=""
            role="presentation"
            style={undefined}
          >
            <a
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1"
            >
              <input
                className="form-control"
                id={undefined}
                onChange={[Function]}
                placeholder="Search..."
                type="text"
                value=""
              />
            </a>
          </li>
          <li
            className="divider"
            onKeyDown={[Function]}
            role="separator"
            style={undefined}
          />
        </ul>
      </li>
    </ul>
    <div
      className="navbar-form navbar-left"
    >
      <button
        className="btn btn-default"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        CLEAR ALL FILTERS
      </button>
    </div>
  </div>
</nav>
`;
